dolphin/task_manager/resources.py

```diff
@@ -37,13 +53,46 @@ class StorageVolumeTask:
     def __init__(self, context, storage_id):
         self.storage_id = storage_id
         self.context = context
+        self.driver = drivermanager.DriverManager()
 
     def sync(self):
-        LOG.info('Volume sync func...')
-        # TODO:
-        # 1. call the driver.list_volumes(context)
-        # 2. update the list_pools info to DB
+
+        try:
+            volume = self.driver.list_volumes(self.context, self.storage_id)
+            volume_id = volume.get('id')
+            db.volume_update(self.context, volume_id, volume)
```

Comment:
Need to check if volume alreay there then update info, otherwise create new entry.
similary we should have a logic to sync DB with newly collected resource_info . ie:- if there is volume in DB but newly collected volume list do not have it.. then we have to delete form our DB.